app/main.py

```
@@ -1,3 +1,80 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union["Distance", int, float]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+    def __mul__(self, other: Union[int, float]) -> "Distance":
+        if not isinstance(other, (int, float)):
+            return NotImplemented
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: Union[int, float]) -> "Distance":
+        if not isinstance(other, (int, float)) or other == 0:
+            return NotImplemented
+
+        rounded_km = round(self.km / other, 2)
+        return Distance(rounded_km)
```

`__truediv__` currently returns `NotImplemented` when `other == 0` (division by zero). Division by zero should raise a `ZeroDivisionError` instead of returning `NotImplemented`. Change the zero check to raise (for example: `if other == 0: raise ZeroDivisionError("division by zero")`) so semantics match normal Python behavior.